library/core/src/mem/maybe_uninit.rs

```diff
@@ -1297,7 +1300,8 @@ impl<T, const N: usize> MaybeUninit<[T; N]> {
     /// let data: [MaybeUninit<u8>; 1000] = MaybeUninit::uninit().transpose();
     /// ```
     #[unstable(feature = "maybe_uninit_uninit_array_transpose", issue = "96097")]
-    pub fn transpose(self) -> [MaybeUninit<T>; N] {
+    #[inline(always)]
```

Comment:
Please just use `#[inline]` for these.  `always` should be avoided without particular justification, so for *this* PR please just use `#[inline]`.

Then we can discuss changing it to `inline(always)` in a subsequent PR if you'd like.

Reply:
I don't like the scope creep anyway, so lemme just pull this out into a seperate PR.